pkg/sql/schema_changer_test.go

```diff
@@ -1245,6 +1245,49 @@ func dropIndexSchemaChange(
 	}
 }
 
+// TestDropColumn tests that dropped columns properly drop their Table's CHECK constraints
+func TestDropColumn(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+	params, _ := tests.CreateTestServerParams()
+
+	s, sqlDB, kvDB := serverutils.StartServer(t, params)
+	defer s.Stopper().Stop(context.TODO())
+
+	if _, err := sqlDB.Exec(`
+CREATE DATABASE t;
+CREATE TABLE t.test (k INT PRIMARY KEY, v INT CHECK (v >= 0), a INT DEFAULT 0 CHECK (a <= v), b INT default 100 CHECK (b > a) );
+`); err != nil {
+		t.Fatal(err)
+	}
+
+	// Read table descriptor.
+	tableDesc := sqlbase.GetTableDescriptor(kvDB, "t", "test")
+	if len(tableDesc.Checks) != 3 {
+		t.Fatalf("Expected 3 checks but got %d ", len(tableDesc.Checks))
+	}
+
+	// Bulk insert.
+	const maxValue = 5000
+	if err := bulkInsertIntoTable(sqlDB, maxValue); err != nil {
+		t.Fatal(err)
+	}
+
+	if _, err := sqlDB.Exec("ALTER TABLE t.test DROP v"); err != nil {
+		t.Fatal(err)
+	}
+
+	// Re-read table descriptor.
+	tableDesc = sqlbase.GetTableDescriptor(kvDB, "t", "test")
+	// Only the check on column c should remain
```

Comment:
I think you renamed column c to something else? In any case, I recommend: "Only check_ab should remain."